components/src/atoms/InlineNotification/MiniInlineNotification.tsx

```diff
@@ -0,0 +1,102 @@
+import { Btn, Flex, Link } from '../../primitives'
+import { ALIGN_CENTER, DIRECTION_ROW, DISPLAY_INLINE_BLOCK } from '../../styles'
+import { SPACING } from '../../ui-style-constants'
+import { LegacyStyledText } from '../StyledText'
+
+import type { MouseEventHandler } from 'react'
+import type { StyleProps } from '../../primitives'
+
+type MiniInlineNotificationType = 'warning' | 'error' | 'success'
+
+export interface MiniInlineNotificationProps extends StyleProps {
+  /** name constant of the icon to display */
+  type: MiniInlineNotificationType
+  /** InlineNotification contents */
+  message: string
+  /** optional handler to show close button/clear alert  */
+  onCloseClick?: (() => void) | MouseEventHandler<HTMLButtonElement>
+  linkText?: string
+  onLinkClick?: (() => void) | MouseEventHandler<HTMLButtonElement>
+  iconName?: string
+  fixedWidth?: boolean
+}
+
+const INLINE_NOTIFICATION_PROPS_BY_TYPE: Record<
+  MiniInlineNotificationType,
+  { color: string; iconName: string; backgroundColor: string }
+> = {
+  warning: {
+    color: COLORS.yellow60,
+    iconName: 'ot-alert',
+    backgroundColor: COLORS.yellow30,
+  },
+  error: {
+    color: COLORS.red60,
+    iconName: 'ot-alert',
+    backgroundColor: COLORS.red30,
+  },
+  success: {
+    color: COLORS.green60,
+    iconName: 'ot-check',
+    backgroundColor: COLORS.green30,
+  },
+}
+
+export function MiniInlineNotification(
+  props: MiniInlineNotificationProps
+): JSX.Element {
+  const { onCloseClick, message, type, linkText, onLinkClick, iconName } = props
+  const inlineNotificationProps = INLINE_NOTIFICATION_PROPS_BY_TYPE[type]
+  return (
+    <Flex
+      alignItems={ALIGN_CENTER}
+      backgroundColor={INLINE_NOTIFICATION_PROPS_BY_TYPE[type].backgroundColor}
+      data-testid={`InlineNotification_${type}`}
+      flexDirection={DIRECTION_ROW}
+      width="100%"
+      minWidth="'12.625rem'"
+      borderRadius={BORDERS.borderRadius4}
+      paddingY={SPACING.spacing4}
+      paddingRight={SPACING.spacing4}
+      paddingLeft={SPACING.spacing8}
+      gridGap={SPACING.spacing8}
+      {...props}
+    >
+      <Flex gridGap={SPACING.spacing8} alignItems={ALIGN_CENTER} flex="1">
+        <Flex>
+          {/* If icon is not wrapped in a Flex component, it shrinks. */}
+          <Icon
+            name={iconName ?? inlineNotificationProps.iconName}
+            size={SPACING.spacing16}
+            color={inlineNotificationProps.color}
+          />
+        </Flex>
+        <LegacyStyledText as="label" display={DISPLAY_INLINE_BLOCK}>
+          {message}{' '}
+          {linkText && onLinkClick && (
+            <Link onClick={onLinkClick} textDecoration="underline">
+              {linkText}
+            </Link>
+          )}
+        </LegacyStyledText>
+      </Flex>
+      {onCloseClick && (
+        <Btn
+          data-testid="InlineNotification_close-button"
+          onClick={onCloseClick}
+          width="28px"
+          height="28px"
```

Comment:
basically we use `rem` except image.

```suggestion
          width="1.75rem"
          height="1.75rem"
```